language/tools/read-write-set/dynamic/src/dynamic_analysis.rs

```diff
@@ -257,13 +262,22 @@ pub fn bind_formals<R: MoveResolver>(
         .map_err(|_| anyhow!("Failed to get module from storage"))?
         .ok_or_else(|| anyhow!("Failed to get module"))?;
 
-    let func_type = Function::new_from_name(&compiled_module, fun)
-        .ok_or_else(|| anyhow!("Failed to find function"))?
+    let func_sig = Function::new_from_name(&compiled_module, fun)
+        .ok_or_else(|| anyhow!("Failed to find function"))?;
+
+    // Return empty access when we have a malformed entry.
+    if func_sig.parameters.len() != actuals.len() + signers.len()
+        || func_sig.type_parameters.len() != type_actuals.len()
+    {
+        return Ok(ConcretizedFormals::empty());
```

Comment:
I think this should be an error rather than succeeding silently. It would be really hard to discover an error where you (e.g.) pass the wrong number of actuals...

Reply:
IMHO returning the empty writeset should be the right behavior from the caller side: when we have a script with mismatch number of arguments, we will just run the prologue and epilogue. So empty writeset is wha'ts expected here. Should I repeat this check on the caller side (in diem side when we analyze the transaction)?

Reply:
Since this is meant to be run online returning an empty writeset here makes sense to me.

You could maybe have some way to short circuit the prologue/epilogue if this fails, but this would then mean you couldn't charge gas for this work